airbyte_cdk/sources/declarative/parsers/model_to_component_factory.py

```diff
@@ -2868,8 +2875,59 @@ def create_oauth_authenticator(
             message_repository=self._message_repository,
             profile_assertion=profile_assertion,
             use_profile_assertion=model.use_profile_assertion,
+            refresh_token_error_status_codes=refresh_token_error_status_codes,
+            refresh_token_error_key=refresh_token_error_key,
+            refresh_token_error_values=refresh_token_error_values,
         )
 
+    @staticmethod
+    def _get_refresh_token_error_information(
+        model: OAuthAuthenticatorModel,
+    ) -> Tuple[Tuple[int, ...], str, Tuple[str, ...]]:
+        """
+        In a previous version of the CDK, the auth error as config_error was only done if a refresh token updater was
+        defined. As a transition, we added those fields on the OAuthAuthenticatorModel. This method ensures that the
+        information is defined only once and return the right fields.
+        """
+        refresh_token_updater = model.refresh_token_updater
+        is_defined_on_refresh_token_updated = refresh_token_updater and (
+            refresh_token_updater.refresh_token_error_status_codes
+            or refresh_token_updater.refresh_token_error_key
+            or refresh_token_updater.refresh_token_error_values
+        )
+        is_defined_on_oauth_authenticator = (
+            model.refresh_token_error_status_codes
+            or model.refresh_token_error_key
+            or model.refresh_token_error_values
+        )
+        if is_defined_on_refresh_token_updated and is_defined_on_oauth_authenticator:
+            raise ValueError(
+                "refresh_token_error should either be defined on the OAuthAuthenticatorModel or the RefreshTokenUpdaterModel, not both"
+            )
+
+        if is_defined_on_refresh_token_updated:
+            not_optional_refresh_token_updater: RefreshTokenUpdaterModel = refresh_token_updater  # type: ignore  # we know from the condition that this is not None
+            return (
+                tuple(not_optional_refresh_token_updater.refresh_token_error_status_codes)
```

Comment:
Note that this will have the same behavior as the previous code but is a bit different. The previous code would pass `refresh_token_updater.refresh_token_error_status_codes` directly which felt very dangerous because the typing was different (`Tuple[int, ...]` vs `Optional[List[int]]`). I fear that the previous code could lead to None exception [here](https://github.com/airbytehq/airbyte-python-cdk/blob/650414816c04c644bff4ace1e4ac1408556b155e/airbyte_cdk/sources/streams/http/requests_native_auth/abstract_oauth.py#L182) if `refresh_token_error_status_codes` was not defined. I don't have signals to believe this actually happen in prod though...